apis/refs/v1beta1/gcpserviceaccountref.go

```diff
@@ -14,20 +14,111 @@
 
 package v1beta1
 
-type MetricsGcpServiceAccountRef struct {
-	/* The Email of the Google Cloud Service Account (GSA) used for exporting Config Sync metrics to Cloud Monitoring. The GSA should have the Monitoring Metric Writer(roles/monitoring.metricWriter) IAM role. The Kubernetes ServiceAccount `default` in the namespace `config-management-monitoring` should be bound to the GSA. Allowed value: The `email` field of an `IAMServiceAccount` resource. */
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type IAMServiceAccountRef struct {
+	/* The GCP Service Account Email used for auth when secretType is gcpServiceAccount. Allowed value: The `email` field of an `IAMServiceAccount` resource. */
```

Comment:
I'm wondering if we want this documentation to be on the field in the parent type, rather than having different references for each.  I think it might also make our docs a bit simpler if we just linked to documentation on how to use refs, rather than repeating it every time.

Reply:
+1 to have the doc in the parent type.  Also, I'm wondering if we do need two SA resolve methods here (Maybe one is better?)

Reply:
This document can be on the parent type as it is generic. The one on  line 39 can't as it is field specific.

Reply:
I proposed we merge this refactor, then look at de-duping the two types, by seeing whether it makes sense when we move documentation to the parent.

We need two SA resolve types if we have two types.  The bulk of the logic is shared between them though, they both call into resolveServiceAccount.